src/Tests/Microsoft.NET.Build.Tests/GivenThatWeHaveAPackageReferenceWithAliases.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeHaveAPackageReferenceWithAliases : SdkTest
+    {
+
+        public GivenThatWeHaveAPackageReferenceWithAliases(ITestOutputHelper log) : base(log)
+        { }
+
+        [Fact]
+        public void CanBuildProjectWithPackageReferencesWithConflictingTypes()
+        {
+            var targetFramework = "netcoreapp3.0";
+            var packageReferences = GetPackageReferencesWithConflictingTypes(targetFramework, "A", "B");
+
+            TestProject testProject = new TestProject()
+            {
+                Name = "Project",
+                IsSdkProject = true,
+                IsExe = false,
+                TargetFrameworks = targetFramework,
+            };
+
+            testProject.PackageReferences.Add(packageReferences.First());
+            testProject.PackageReferences.Add(
+                new TestPackageReference(
+                    packageReferences.Last().ID,
+                    packageReferences.Last().Version,
+                    packageReferences.Last().NupkgPath,
+                    packageReferences.Last().PrivateAssets,
+                    aliases: "Special"));
+            var packagesPaths = packageReferences.Select(e => Path.GetDirectoryName(e.NupkgPath));
+
+            testProject.AdditionalProperties.Add("RestoreAdditionalProjectSources",
+                                     "$(RestoreAdditionalProjectSources);" + string.Join(";", packagesPaths));
+
+            //  Use a test-specific packages folder
+            testProject.AdditionalProperties["RestorePackagesPath"] = @"$(MSBuildProjectDirectory)\..\pkg";
```

Comment:
will this drop a temp restore file in a share folder or a folder specific for this test

Reply:
Folder specific for this test.